doc: posix: mark putmsg as supported

moonlight83340 opened the pull request:
`putmsg()` is missing in the documentation.
Has been discuss here https://github.com/zephyrproject-rtos/zephyr/pull/67711

moonlight83340 committed:
doc: posix: mark putmsg as supported

Mark `putmsg()` as implemented but will fail with ``ENOSYS``
Was missing on the documentation before.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

nashif commented on doc/services/portability/posix/option_groups/index.rst:
```diff
@@ -509,7 +509,7 @@ _XOPEN_STREAMS
     ioctl(),yes
     isastream(),
-    putmsg(),
+    putmsg(),yes
```
This is not correct, putmsg was not fully implemented and we just provide an empty stub, can we clarify please, otherwise it is false advertisement, "yes" should be used only for fully implemented functions IMO, for empty stubs, lets use something else to make it clear.

cfriedt replied:
Yes - good catch.

@moonlight83340 - can you please match what was done for other functions where ENOSYS is reported?

Also, in a second commit, please ensure that other ENOSYS functions have similar entries in the docs.

moonlight83340 replied:
in posix directory, I saw we still put the yes in the documentation even when there are ENOSYS 😓 
Is there any other part of the documentation that should be modify ? after a quick search for ENOSYS functions and documentation files, I'm not seing others changes.

cfriedt replied:
Please copy the .rst source where it says 

"(will fail with ENOSYS†)"

https://docs.zephyrproject.org/latest/services/portability/posix/conformance/index.html#posix-system-interfaces

moonlight83340 replied:
should I keep the yes with the "(will fail with ENOSYS†)" in the compliance documentation part ?

moonlight83340 replied:
> This is not correct, putmsg was not fully implemented and we just provide an empty stub, can we clarify please, otherwise it is false advertisement, "yes" should be used only for fully implemented functions IMO, for empty stubs, lets use something else to make it clear.

Do you think modifications are okay ? 😅

cfriedt commented on doc/services/portability/posix/option_groups/index.rst:
```diff
@@ -509,7 +509,7 @@ _XOPEN_STREAMS
     ioctl(),yes
     isastream(),
-    putmsg(),
+    putmsg(),yes
```
```suggestion
    putmsg(), yes (will fail with ``ENOSYS``:ref:`†<posix_undefined_behaviour>`)
```

cfriedt commented on doc/services/portability/posix/conformance/index.rst:
```diff
@@ -109,7 +109,7 @@ POSIX System Interfaces
     _XOPEN_REALTIME, -1,
     _XOPEN_REALTIME_THREADS, -1,
-    :ref:`_XOPEN_STREAMS<posix_option_xopen_streams>`, -1, :kconfig:option:`CONFIG_NET_SOCKETS`
+    :ref:`_XOPEN_STREAMS<posix_option_xopen_streams>`, -1, :kconfig:option:`CONFIG_NET_SOCKETS` (will fail with ``ENOSYS``:ref:`†<posix_undefined_behaviour>`)
```
Once all features for `_XOPEN_STREAMS` are available (at the interface level), this can be changed to `200809L` instead of `-1`. But the reference to `CONFIG_NET_SOCKETS` is invalid, so please remove it.

```suggestion
    :ref:`_XOPEN_STREAMS<posix_option_xopen_streams>`, -1, :ref:`†<posix_undefined_behaviour>`
```

moonlight83340 commented:
Also in lib/posix/options/sched.c , all the function are marks yes but no mention of enosys. Should it be the same as here ?
I have done my self https://github.com/zephyrproject-rtos/zephyr/pull/68362 a modification on yes and now we discuss about it it seems it should have the mention enosys also.

cfriedt commented:
> Possibily have to modify the documentation of :
> https://github.com/zephyrproject-rtos/zephyr/pull/68200
> https://github.com/zephyrproject-rtos/zephyr/pull/68316

I think the above two are all that are necessary, the last PR looks like it does mention ENOSYS

moonlight83340 commented:
Could be good to pass the documentation precision before 3.6 is released, should I chnage the PR and add the others modifications or do anothers PR for the others documentation modification ?

cfriedt commented:
> Should I do it in a different PR ?

@moonlight83340 - one pr should be fine

moonlight83340 committed:
doc: posix: sched functions will fail with ENOSYS

Mark `sched_setparam()` , `sched_setscheduler()`
and `sched_rr_get_interval()`
have implemented but will fail with ``ENOSYS``
Were missing on the documentation before.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 commented:
I should change the PR name for something else, I think..
